module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,258 @@
+        int fd = socket(addr_ptr->ai_family, addr_ptr->ai_socktype, addr_ptr->ai_protocol);
+
+        if (fd == -1) {
+            // Bad fd
+            continue;
+        }
+        else if (connect(fd, addr_ptr->ai_addr, addr_ptr->ai_addrlen) != -1) {
+            // Connection successful
+            freeaddrinfo(address);
+            return fd;
+        }
+        // Connection was not successful
+        close(fd);
+    }
+
+    // No connection was successful
+    freeaddrinfo(address);
+    log<NUClear::ERROR>(fmt::format("Cannot connect to server: {}:{}", server_name, port));
+    return -1;
+}
+
+Webots::Webots(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)) {
+    on<Trigger<GlobalConfig>, Configuration>("webots.yaml").then([this](const GlobalConfig& global_config, const Configuration& local_config) {
+        // Use configuration here from file webots.yaml
+
+        // clang-format off
+        auto lvl = local_config["log_level"].as<std::string>();
+        if (lvl == "TRACE") { this->log_level = NUClear::TRACE; }
+        else if (lvl == "DEBUG") { this->log_level = NUClear::DEBUG; }
+        else if (lvl == "INFO") { this->log_level = NUClear::INFO; }
+        else if (lvl == "WARN") { this->log_level = NUClear::WARN; }
+        else if (lvl == "ERROR") { this->log_level = NUClear::ERROR; }
+        else if (lvl == "FATAL") { this->log_level = NUClear::FATAL; }
+        // clang-format on
+
+        int fd = tcpip_connect(local_config["server_address"].as<std::string>(), local_config["port"].as<std::string>());
+
+        // Tell webots who we are
+        send_player_details(fd, global_config);
+
+        on<IO>(fd, IO::READ | IO::WRITE | IO::ERROR | IO::CLOSE).then([this, fd]() {
+            // Receiving
+
+            // Get the size of the message
+            uint32_t Nn;
+            if (recv(fd, &Nn, sizeof(Nn), 0 != sizeof(Nn))) {
+                log<NUClear::ERROR>("Failed to read message size from TCP connection");
+                return;
+            }
+
+            uint32_t Nh = ntohl(Nn);  // Convert from network endian to host endian
+
+            // Get the message
+            std::vector<char> data(Nh, 0);
+            if (uint64_t(recv(fd, data.data(), Nh, 0)) != Nh) {
+                log<NUClear::ERROR>("Failed to read message from TCP connection");
+                return;
+            }
+
+            // Deserialise the message into a neutron
+            SensorMeasurements msg = NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(data);
+            
+            translate_and_emit_sensor(msg);
+        });
+
+        on<Every<1, std::chrono::seconds>>().then([this, fd]() {
+            // Sending
+            std::vector<char> data = NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(to_send);
+            uint32_t N             = htonl(data.size());
+            send(fd, &N, sizeof(N), 0);
+            send(fd, data.data(), N, 0);
```

Comment:
Should we keep this consistent?
```suggestion
            uint32_t Nn             = htonl(data.size());
            send(fd, &Nn, sizeof(Nn), 0);
            send(fd, data.data(), Nn, 0);
```